gfx.cpp

```diff
@@ -9414,13 +9525,40 @@ class GfxInternal
         window_width_  = window_width;
         window_height_ = window_height;
         HRESULT const hr = swap_chain_->ResizeBuffers(max_frames_in_flight_, window_width, window_height, back_buffer_format_, DXGI_SWAP_CHAIN_FLAG_ALLOW_TEARING);
+        if(hr == DXGI_ERROR_DEVICE_REMOVED || hr == DXGI_ERROR_DEVICE_RESET || hr == DXGI_ERROR_DEVICE_HUNG)
+        {
+            GFX_TRY(handleDeviceLost());
+        }
+        else if(FAILED(hr))
+        {
+            return GFX_SET_ERROR(kGfxResult_InternalError, "Error detected during resizeBuffers: %s", hr);
+        }
         fence_index_ = swap_chain_->GetCurrentBackBufferIndex();
         GFX_TRY(acquireSwapChainBuffers());
-        if(!SUCCEEDED(hr))
-            return GFX_SET_ERROR(kGfxResult_InternalError, "Unable to resize the swap chain buffers");
         GFX_TRY(createBackBufferRTVs());
         return kGfxResult_NoError;
     }
+
+    GfxResult handleDeviceLost()
+    {
+        HRESULT const reason = device_->GetDeviceRemovedReason();
+        LPSTR error_text = nullptr;
+        DWORD const result = FormatMessageA(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ALLOCATE_BUFFER, nullptr, reason,
+                MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPSTR>(&error_text), 0, nullptr);
+        if (command_list_ != nullptr)
+        {
+            command_list_->Release();
+            command_list_ = nullptr;
+        }
```

Comment:
Need to explicitly close the command list here as otherwise will get failures later on even if just terminating